samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -618,106 +623,45 @@ struct find_valid_bis_data {
 /**
  * This is called for each BIS in a subgroup
  *
- * It returns `false` if the current BIS contains all of the channels we are looking for,
- * or if it does not contain any and we are looking for BT_AUDIO_LOCATION_MONO_AUDIO. This stops
- * the iteration of the remaining BIS in the subgroup.
- *
- * It returns `true` if the BIS either contains none or some of the channels we are looking for.
- * If it contains some, then that is being stored in the user_data, so that the calling function
- * can check if a combination of the BIS satisfy the channel allocations we want.
+ * Gets BIS channel allocation (if exists).
+ * Always returns `true` to continue to next BIS
  */
-static bool find_valid_bis_cb(const struct bt_bap_base_subgroup_bis *bis,
-					       void *user_data)
+static bool bis_get_channel_allocation_cb(const struct bt_bap_base_subgroup_bis *bis,
+					  void *user_data)
 {
-	struct find_valid_bis_data *data = user_data;
-	struct bt_audio_codec_cfg codec_cfg = {0};
-	enum bt_audio_location chan_allocation;
-	int err;
+	struct bis_channel_allocation_data *data = user_data;
+	struct bt_audio_codec_cfg codec_cfg;
+	int err, idx;
+
+	idx = data->cnt++;
+	data->bis[idx].index = bis->index;
+	data->bis[idx].chan_allocation_available = false;
 
 	err = bt_bap_base_subgroup_bis_codec_to_codec_cfg(bis, &codec_cfg);
 	if (err != 0) {
 		printk("Could not get codec configuration for BIS: %d\n", err);
-		return true;
+
+		return true; /* continue to next BIS */
 	}
 
-	err = bt_audio_codec_cfg_get_chan_allocation(&codec_cfg, &chan_allocation, true);
+	/* Only LC3 codec supported (assumed at this point) */
+	codec_cfg.id = BT_HCI_CODING_FORMAT_LC3;
```

Comment:
```suggestion
	/* Only LC3 codec supported (assumed at this point). 
	 * It is required to set the codec_cfg.id to use the fallback behavior of bt_audio_codec_cfg_get_chan_allocation. 
	 */
	codec_cfg.id = BT_HCI_CODING_FORMAT_LC3;
```

We could just not use the fallback behavior; up to you :)

Reply:
Yes. Code reduction. I will take that